utils/remote.cc

```diff
@@ -1974,8 +1984,128 @@ static void printGroups(tr_variant* top)
+            {
+                continue;
+            }
+            /*
+            if (tr_variantDictFindInt(d, TR_KEY_eta, &eta) && 
+                tr_variantDictFindInt(d, TR_KEY_leftUntilDone, &leftUntilDone) &&
+                tr_variantDictFindStrView(d, TR_KEY_name, &name) && tr_variantDictFindInt(d, TR_KEY_rateDownload, &down) &&
+                tr_variantDictFindInt(d, TR_KEY_rateUpload, &up) &&
+                tr_variantDictFindInt(d, TR_KEY_sizeWhenDone, &sizeWhenDone) &&
+                && tr_variantDictFindReal(d, TR_KEY_uploadRatio, &ratio))
+                */
+            bool include = negate;
+            std::string status = getStatusString(d);
+            std::string_view name;
+            switch (filter[pos])
+            {
+            case 'i': // Status = Idle
+                if (status == "Idle")
+                {
+                    include = !include;
+                }
+                break;
+            case 'd': // Downloading (Status is Downloading or Up&Down)
+                if (status.find("Down") != std::string::npos)
+                {
+                    include = !include;
+                }
+                break;
+            case 'u': // Uploading (Status is Uploading, Up&Down or Seeding
+                if ((status.find("Up") != std::string::npos) || (status == "Seeding"))
+                {
+                    include = !include;
+                }
+                break;
+            case 'l': // label
+                tr_variant* l;
+                if (tr_variantDictFindList(d, TR_KEY_labels, &l))
+                {
+                    size_t child_pos = 0;
+                    tr_variant const* child;
+                    std::string_view sv;
+                    while ((child = tr_variantListChild(l, child_pos++)))
+                    {
+                        if (tr_variantGetStrView(child, &sv))
+                        {
+                            if (arg == sv)
+                            {
+                                include = !include;
+                                break;
+                            }
+                        }
+                    }
+                }
+                break;
+            case 'n': // Torrent name substring
+                if (!tr_variantDictFindStrView(d, TR_KEY_name, &name))
+                {
+                    continue;
+                }
+                if (name.find(arg) != std::string::npos)
+                {
+                    include = !include;
+                }
+                break;
+            case 'r': // Minimal ratio
+                double ratio;
+                if (!tr_variantDictFindReal(d, TR_KEY_uploadRatio, &ratio))
+                {
+                    continue;
+                }
+                if (ratio >= std::stof(std::string(arg)))
+                {
+                    include = !include;
+                }
+                break;
```

Comment:
I'm surprised that the compiler lets you declare a local like this after the case statement...? I didn't think that was legal.

Anyway, the scope of `ratio` can be reduced here by using an if decl block:

```suggestion
                double ratio;
                if (double ratio = {}; !tr_variantDictFindReal(d, TR_KEY_uploadRatio, &ratio))
                {
                    continue;
                }
                else if (ratio >= std::stof(std::string(arg)))
                {
                    include = !include;
                }
                break;
```